core-bundle/src/Resources/contao/modules/ModuleSearch.php

```diff
@@ -321,13 +325,69 @@ protected function compile()
 					$objTemplate->hasContext = true;
 				}
 
+				$this->addImageToTemplateFromSearchResult($arrResult[$i], $objTemplate);
+
 				$this->Template->results .= $objTemplate->parse();
 			}
 
 			$this->Template->header = vsprintf($GLOBALS['TL_LANG']['MSC']['sResults'], array($from, $to, $count, $strKeywords));
 			$this->Template->duration = System::getFormattedNumber($query_endtime - $query_starttime, 3) . ' ' . $GLOBALS['TL_LANG']['MSC']['seconds'];
 		}
 	}
+
+	protected function addImageToTemplateFromSearchResult(array $result, Template $template): void
+	{
+		$template->hasImage = false;
+
+		if (!isset($result['meta']) || null == $result['meta'])
+		{
+			return;
+		}
+
+		$meta = json_decode($result['meta'], true);
+
+		// Reverse so the last element wins
+		$meta = array_reverse($meta);
+
+		foreach ($meta as $v)
+		{
+			if (!isset($v['https://schema.org/primaryImageOfPage']))
+			{
+				continue;
+			}
+
+			$figureBuilder = $this->getFigureBuilderIfResourceExists($v['https://schema.org/primaryImageOfPage']['contentUrl']);
+
+			if (null === $figureBuilder)
+			{
+				continue;
+			}
+
+			$figureMeta = array();
+			if (isset($v['https://schema.org/primaryImageOfPage']['caption']))
+			{
+				$figureMeta[Metadata::VALUE_CAPTION]  = $v['https://schema.org/primaryImageOfPage']['caption'];
+			}
+			if (isset($v['https://schema.org/primaryImageOfPage']['name']))
+			{
+				$figureMeta[Metadata::VALUE_TITLE]  = $v['https://schema.org/primaryImageOfPage']['name'];
+			}
+
+			if (isset($v['https://schema.org/primaryImageOfPage']['alternateName']))
+			{
+				$figureMeta[Metadata::VALUE_ALT]  = $v['https://schema.org/primaryImageOfPage']['alternateName'];
+			}
+
+			$template->hasImage = true;
+			$template->image = (object) $figureBuilder
+				->setSize($this->imgSize)
+				->setMetadata(new Metadata($figureMeta))
+				->build()
+				->getLegacyTemplateData();
+
+			return;
```

Comment:
Let's add the built `Figure` to the template as well (like in https://github.com/contao/contao/pull/2734).

Reply:
Done.